code/modules/unit_tests/organs.dm

```diff
@@ -0,0 +1,174 @@
+		// Skip prototypes.
+		if(organ_type in TEST_ORGAN_BLACKLIST)
+			continue
+
+		// Appropriate mob (Human) which will receive organ.
+		var/mob/living/carbon/human/lab_rat = allocate(/mob/living/carbon/human/consistent)
+		var/obj/item/organ/test_organ = new organ_type()
+
+		// Inappropriate mob (Dog) which will hopefully reject organ.
+		var/mob/living/basic/pet/dog/lab_dog = allocate(/mob/living/basic/pet/dog/corgi)
+		var/obj/item/organ/reject_organ = new organ_type()
+
+		// Insert organ and store status code in var.
+		var/inserted_ok = test_organ.Insert(lab_rat, special = TRUE, drop_if_replaced = FALSE)
+		// Ensure organ was rejected.
+		var/rejected_ok = !reject_organ.Insert(lab_dog, special = TRUE, drop_if_replaced = FALSE)
+
+		// For the Human, expects status code 1
+		if(!inserted_ok)
+			TEST_FAIL("The organ \"[test_organ.type]\" was not inserted in the mob when expected, Insert() returned FALSE when TRUE was expected.")
+			continue
+
+		// For the Dog, expects status code 0 or falsy value.
+		if(!rejected_ok)
+			TEST_FAIL("The \"[test_organ.type]\" was inserted into a basic mob (Corgi) when it wasn't expected.")
+
+		// Inserting Nightmare brain causes the Human's species to change.
+		// Species change swaps out all the organs, making test_organ un-usable by this point.
+		if(test_organ.type == /obj/item/organ/internal/brain/shadow/nightmare)
+			continue
+
+		// Check vars on Human and organ, they are expected to be present after Insert().
+		if(!TEST_ORGAN_INSERTED_OK(lab_rat, test_organ))
+			TEST_FAIL("The organ \"[test_organ.type]\" was not properly inserted in the mob, some variables were not assigned when expected.")
+			continue
+
+		// Now yank it back out.
+		test_organ.Remove(lab_rat, special = TRUE)
+
+		// Check vars on Human and organ, they are expected to be deleted after Remove().
+		if(!TEST_ORGAN_REMOVED_OK(lab_rat, test_organ))
+			TEST_FAIL("The organ \"[test_organ.type]\" was not properly removed from the mob, some variables were not reset when expected.")
+
+#undef TEST_ORGAN_INSERTED_OK
+#undef TEST_ORGAN_REMOVED_OK
+#undef TEST_ORGAN_BLACKLIST
+
+//-- Sanity: DNA INFUSER ORGANS / ORGAN SET BONUS --//
+/// Ensures the developers properly change IDs to be unique.
+/datum/unit_test/organ_sanity/organ_set_bonus_id/Run()
+	var/list/bonus_effects = typesof(/datum/status_effect/organ_set_bonus)
+	var/list/existing_ids = list()
+	for(var/datum/status_effect/organ_set_bonus/bonus_effect as anything in bonus_effects)
+		var/effect_id = initial(bonus_effect.id)
+		var/existing_status = (effect_id in existing_ids)
+		if(existing_status)
+			TEST_FAIL("The ID of [bonus_effect] was duplicated in another status effect.")
+		else
+			existing_ids += effect_id
+
+/// Utility proc which searches a mob for a Status Effect using a given typepath.
+/datum/unit_test/organ_sanity/organ_set_bonus_sanity/proc/check_status_type(mob/living/carbon/human/lab_rat, datum/status_effect/status_type)
```

Comment:
This whole proc doesn't need to exist and can instead be `locate(status_type) in lab_rat.status_effects`.